### PR TITLE
Lecturer and student site 2

### PR DESCRIPTION
**Pull request by LucRome**
Foundation for lecturer and student sites

### CODE REVIEW

File: eCourse_backend/courses/urls.py [hunk 1]
```diff
@@ -22,11 +22,13 @@
     path('delete/<id>', delete_course, name='delte_course'),
     # course
     path('overview/page<int:page>', course_overview, name='course_overview'),
-    # create user iframes
+    # create course iframes
     path('iframes/studentlist/<id>', course_student_list_iframe,
          name='course_studentlist_iframe'),
     # admin: create course
-    path('create/', create_course, name="create_course"),
+    path('create/', create_course_admin, name="create_course"),
```
Comment (phie73) [inline eCourse_backend/courses/urls.py]: please don't change function names. The backend tests depend on this names.

File: eCourse_backend/courses/views.py [hunk 2]
```diff
@@ -15,41 +15,53 @@
...
-    context = {
-        'filter_form': filter_form,
-        'lecturer': lecturer,
-        'students': students,
-    }
-    return render(request, 'courses/iframes/course_student_list.html', context)
+        # files
+        files = dir()
+        for e in exercise:
+            print(e.id)
```
Comment (phie73) [inline eCourse_backend/courses/views.py]: please remove debug info

File: eCourse_backend/courses/views.py [hunk 3]
```diff
@@ -15,41 +15,53 @@
...
+        return render(
+            request,
+            'courses/iframes/course_student_list.html',
+            context)
+
+    # student
+    if (request.user.type == 3):
+        # exercises
+        exercise = Exercise.objects.filter(course_id=id)
+        print(exercise)
```
Comment (phie73) [inline eCourse_backend/courses/views.py]: please remove debug info

File: eCourse_backend/courses/views.py [hunk 4]
```diff
@@ -113,7 +125,7 @@ def view_course(request, id):
 
 @login_required
 @permission_required('courses.add_course', raise_exception=True)
-def create_course(request):
+def create_course_admin(request):
```
Comment (phie73) [inline eCourse_backend/courses/views.py]: see urls.py

File: eCourse_backend/courses/views.py [hunk 5]
```diff
@@ -143,15 +164,28 @@ def delete_course(request, id):
 
 
 @login_required
-@permission_required('course.change_course', raise_exception=True)
+@permission_required('courses.change_course', raise_exception=True)
 def edit_course(request, id):
+    updata_success = False
```
Comment (phie73) [inline eCourse_backend/courses/views.py]: please change to update_success

File: eCourse_backend/eCourse_backend/views.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from django.shortcuts import render, get_object_or_404
+from django.contrib.auth.decorators import permission_required, login_required
```
Comment (phie73) [inline eCourse_backend/eCourse_backend/views.py]: please remove unneeded imports line 1 and permission_requierd

**Review by phie73 — CHANGES_REQUESTED** [review]
looks good overall. Please address  the comments.

File: eCourse_backend/courses/urls.py (hunk 1 above)
Comment (LucRome) [reply eCourse_backend/courses/urls.py]: Should be resolved

File: eCourse_backend/courses/views.py (hunk 2 above)
Comment (LucRome) [reply eCourse_backend/courses/views.py]: should be resolved

File: eCourse_backend/eCourse_backend/views.py (hunk 6 above)
Comment (LucRome) [reply eCourse_backend/eCourse_backend/views.py]: should be resolved

File: eCourse_backend/courses/views.py (hunk 3 above)
Comment (LucRome) [reply eCourse_backend/courses/views.py]: should be resolved

File: eCourse_backend/courses/views.py (hunk 4 above)
Comment (LucRome) [reply eCourse_backend/courses/views.py]: should be resolved

File: eCourse_backend/courses/views.py (hunk 5 above)
Comment (LucRome) [reply eCourse_backend/courses/views.py]: should be resolved

**Review by codingHahn — APPROVED** [review]
Backend side looks good now!

**Review by phie73 — APPROVED** [review]

**Pull request merged**